### PR TITLE
Routes avatar + fixed Vittorias mystical bug

### PR DESCRIPTION
**Pull request by kathinka**
- Fix routes so we can have multiple components on the same page using the same ID. 
- small designadjustment to battery to make it easier for the user to see their input. we can prob omit the slider )
- Fixed issue with previous battery not working du to apiroute being fetched before fetching Id, also added missing const for token who is neccessary to be allowed to verify and then talk to db
- added name to avatar component as those two are tightly connected.
- restructured avatar functions and image array in a utils file to keep the user model clean.
- Set up the dash so its just to build the remaining components to make them appear.

### CODE REVIEW

**Comment by kathinka** [comment]
https://ego-app.onrender.com/ check it out here. we can discuss colorchange on battery :)

File: frontend/src/components/BatterySlider.jsx [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+      <BatterySliderWrapper id={id}>
+        <BatterySVG fillPercentage={fillPercentage} onDrag={handleDrag} />
+        <input
+          type="range"
+          min="0"
+          max="100"
+          value={fillPercentage}
+          onChange={(e) => updateEnergyLevel(e.target.value)}
+        />
+        <div>Current energy level: {fillPercentage}%</div>
```
Comment (vittoriamatteoli) [inline frontend/src/components/BatterySlider.jsx:134]: Yes I think it's smart to say the percentage, Let's see how to implement it in the design

**Review by vittoriamatteoli — COMMENTED** [review]
Thank you Khatinka! I looks great , we will need just to add some missing pieces and we are good to go. :)

**Review by vittoriamatteoli — APPROVED** [review]
Thank you Khatinka! I looks great , we will need just to add some missing pieces and we are good to go. :)

**Review by Yifan-858 — APPROVED** [review]

**Pull request merged**